catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/CreateOperations.java

```diff
@@ -559,4 +474,171 @@ private CreateResponse validateFixCreateResponse(CreateResponse createResponse,
         return createResponse;
     }
 
+    private CreateResponse processPostIngestPlugins(CreateResponse createResponse) {
+        for (final PostIngestPlugin plugin : frameworkProperties.getPostIngest()) {
+            try {
+                createResponse = plugin.process(createResponse);
+            } catch (PluginExecutionException e) {
+                LOGGER.info("Plugin processing failed. This is allowable. Skipping to next plugin.",
+                        e);
+            }
+        }
+        return createResponse;
+    }
+
+    private CreateResponse performRemoteCreate(CreateRequest createRequest,
+            CreateResponse createResponse) {
+        if (!opsCatStoreSupport.isCatalogStoreRequest(createRequest)) {
+            return createResponse;
+        }
+
+        CreateResponse remoteCreateResponse = doRemoteCreate(createRequest);
+        if (createResponse == null) {
+            createResponse = remoteCreateResponse;
+        } else {
+            createResponse.getProperties()
+                    .putAll(remoteCreateResponse.getProperties());
+            createResponse.getProcessingErrors()
+                    .addAll(remoteCreateResponse.getProcessingErrors());
+        }
+        return createResponse;
+    }
+
+    private CreateResponse getCreateResponse(CreateRequest createRequest) throws IngestException {
+        // Call the create on the catalog
+        LOGGER.debug("Calling catalog.create() with {} entries.",
+                createRequest.getMetacards()
+                        .size());
+        if (!Requests.isLocal(createRequest)) {
+            return null;
+        }
+
+        CreateResponse createResponse = sourceOperations.getCatalog()
+                .create(createRequest);
+        createResponse = historian.version(createResponse);
+        return createResponse;
+    }
+
+    private CreateRequest processPreIngestPlugins(CreateRequest createRequest)
+            throws StopProcessingException {
+        for (PreIngestPlugin plugin : frameworkProperties.getPreIngest()) {
+            try {
+                createRequest = plugin.process(createRequest);
+            } catch (PluginExecutionException e) {
+                LOGGER.info("Plugin processing failed. This is allowable. Skipping to next plugin.",
+                        e);
+            }
+        }
+        return createRequest;
+    }
+
+    private CreateRequest processPrecreateAccessPlugins(CreateRequest createRequest)
+            throws StopProcessingException {
+        for (AccessPlugin plugin : frameworkProperties.getAccessPlugins()) {
```

Comment:
Not easily as we're updating the createResponse on each invocation. It would require knowing that the call to `processPreCreate` is purely a side effect method.